app/models/manageiq/providers/autosde/inventory/persister/storage_manager.rb

```diff
@@ -1,29 +1,43 @@
 # This class builds the collections. The parser then adds actual inventory items to the collections.
 # Generally, one collection per application-record class.
+# for more info see: https://www.manageiq.org/docs/guides/providers/persister/inventory_collections
 class ManageIQ::Providers::Autosde::Inventory::Persister::StorageManager < ManageIQ::Providers::Inventory::Persister
   # @return [Hash{Symbol => InventoryRefresh::InventoryCollection}] collections
   attr_reader  :collections
 
   def initialize_inventory_collections
-    # physical storages
-    add_collection(storage, :physical_storages) do |builder|
-      builder.add_default_values(:ems_id => ->(persister) { persister.manager.id })
-      builder.add_properties(
-        :model_class => ManageIQ::Providers::Autosde::StorageManager::PhysicalStorage
-      )
-    end
+    collect_san_addresses
+    collect_host_initiators
+    collect_volume_mappings
+    collect_wwpn_candidates
+    collect_host_initiator_groups
+    collect_physical_storages
+    collect_physical_storage_families
+    collect_storage_resources
+    collect_storage_services
+    collect_cloud_volumes
+  end
 
-    # storage resources
-    add_collection(storage, :storage_resources) do |builder|
-      builder.add_default_values(
-        :ems_id => ->(persister) { persister.manager.id }
-      )
-      builder.add_properties(
-        :parent_inventory_collections => [:physical_storages]
-      )
-    end
+  def strategy
+    nil
+  end
+
+  def parent
+    manager.presence
+  end
 
-    # san addresses
+  def shared_options
+    {
+      :strategy => strategy,
+      :targeted => targeted?,
+      :parent   => parent
+    }
+  end
+
+  protected
+
+  # san addresses
+  def collect_san_addresses
     add_collection(storage, :san_addresses) do |builder|
```

Comment:
Not for this PR but most of these could move to the core inventory collection https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/inventory/persister/builder/storage_manager.rb

If all you are doing is setting the `ems_id`, `model_class`, and `parent_inventory_collections` that actually isn't provider specific but would be the same for all storage_managers.